frontend/app/src/hooks/use-organizations.ts

```diff
@@ -0,0 +1,239 @@
+  );
+
+  const hasOrganizations = useMemo(() => {
+    return (organizationListQuery.data?.rows?.length || 0) > 0;
+  }, [organizationListQuery.data?.rows]);
+
+  const acceptOrgInviteMutation = useMutation({
+    mutationKey: ['organization-invite:accept'],
+    mutationFn: async (data: { inviteId: string }) => {
+      await cloudApi.organizationInviteAccept({
+        id: data.inviteId,
+      });
+    },
+    onError: handleApiError,
+  });
+
+  const rejectOrgInviteMutation = useMutation({
+    mutationKey: ['organization-invite:reject'],
+    mutationFn: async (data: { inviteId: string }) => {
+      await cloudApi.organizationInviteReject({
+        id: data.inviteId,
+      });
+    },
+    onError: handleApiError,
+  });
+
+  const createTenantMutation = useMutation({
+    mutationFn: async (data: {
+      organizationId: string;
+      name: string;
+      slug: string;
+    }) => {
+      const result = await cloudApi.organizationCreateTenant(
+        data.organizationId,
+        {
+          name: data.name,
+          slug: data.slug,
+        },
+      );
+      return result.data;
+    },
+    onError: handleApiError,
+  });
+
+  const cancelInviteMutation = useMutation({
+    mutationFn: async (data: { inviteId: string }) => {
+      await cloudApi.organizationInviteDelete(data.inviteId);
+    },
+    onError: handleApiError,
+  });
+
+  const handleCancelInvite = useCallback(
+    (
+      invite: any,
+      onSuccess: () => void,
+      onOpenChange: (open: boolean) => void,
+    ) => {
+      if (invite) {
+        cancelInviteMutation.mutate(
+          { inviteId: invite.metadata.id },
+          {
+            onSuccess: () => {
+              onSuccess();
+              onOpenChange(false);
+            },
+            onError: () => {
+              onOpenChange(false);
+            },
+          },
+        );
+      }
+    },
+    [cancelInviteMutation],
+  );
+
+  const createTokenMutation = useMutation({
+    mutationFn: async (data: {
+      organizationId: string;
+      name: string;
+      duration: ManagementTokenDuration;
+    }) => {
+      const result = await cloudApi.managementTokenCreate(data.organizationId, {
+        name: data.name,
+        duration: data.duration,
+      });
+      return result.data;
+    },
+    onError: handleApiError,
+  });
+
+  const deleteMemberMutation = useMutation({
+    mutationFn: async (data: { memberId: string; email: string }) => {
+      await cloudApi.organizationMemberDelete(data.memberId, {
+        emails: [data.email],
+      });
+    },
+    onError: handleApiError,
+  });
+
+  const deleteTokenMutation = useMutation({
+    mutationFn: async (data: { tokenId: string }) => {
+      await cloudApi.managementTokenDelete(data.tokenId);
+    },
+    onError: handleApiError,
+  });
+
+  const handleCreateToken = useCallback(
+    (
+      organizationId: string,
+      name: string,
+      duration: ManagementTokenDuration,
+      onSuccess: (data: any) => void,
```

Comment:
this `data` also should have a type, which I _think_ your IDE should be able to infer. it should be the return type of the `createTokenMutation`, which should be generated in that api file